prow/scripts/cluster-integration/kyma-gke-upgrade.sh

```diff
@@ -68,463 +79,399 @@ export EXTERNAL_SOLUTION_TEST_RELEASE_NAME="${EXTERNAL_SOLUTION_TEST_NAMESPACE}"
+
+  local TEST_INFRA_COMMIT_ID
+  TEST_INFRA_COMMIT_ID=$(cd "$TEST_INFRA_SOURCES_DIR" && git rev-parse --short HEAD)
+
+  local IMAGE_REPO_NAME="$REPO_NAME"
+  if [[ "$IMAGE_REPO_NAME" == "test-infra" ]]; then
+    IMAGE_REPO_NAME="kyma"
+  fi
+
+  KYMA_INSTALLER_IMAGE="${DOCKER_PUSH_REPOSITORY}${DOCKER_PUSH_DIRECTORY}/gke-upgradeability/${REPO_OWNER}/${IMAGE_REPO_NAME}:KYMA-${KYMA_COMMIT_ID:-unknown}-TI-${TEST_INFRA_COMMIT_ID:-unknown}"
+
+  (
+    set -x
+    kyma upgrade \
+      --ci \
+      --source local \
+      --src-path "${KYMA_SOURCES_DIR}" \
+      --custom-image "${KYMA_INSTALLER_IMAGE}" \
+      --components "${KYMA_SOURCES_DIR}/installation/resources/installer-cr-cluster.yaml.tpl" \
+      --timeout "${KYMA_UPDATE_TIMEOUT}"
+  )
 }
 
-remove_addons_if_necessary() {
-  tdWithAddon=$(kubectl get td --all-namespaces -l testing.kyma-project.io/require-testing-addon=true -o custom-columns=NAME:.metadata.name --no-headers=true)
+function upgradeKymaToCommit() {
+  COMMIT_ID=$(cd "$KYMA_SOURCES_DIR" && git rev-parse HEAD)
+  log::banner "Updating Kyma with timeout ${KYMA_UPDATE_TIMEOUT} to Kyma commit: ${COMMIT_ID}"
+  (
+    set -x
+    kyma upgrade \
+      --ci \
+      --source "${COMMIT_ID}" \
+      --timeout "${KYMA_UPDATE_TIMEOUT}"
+  )
+}
 
-  if [ -z "$tdWithAddon" ]
-  then
-      echo "- Removing ClusterAddonsConfiguration which provides the testing addons"
-      removeTestingAddons
-      if [[ $? -eq 1 ]]; then
-        exit 1
-      fi
+function upgradeKyma() {
+  if [[ "$BUILD_TYPE" == "release" ]]; then
+    upgradeKymaToRelease
+  elif [[ "$BUILD_TYPE" == "pr" ]]; then
+    upgradeKymaToCheckedoutVersion
   else
-      echo "- Skipping removing ClusterAddonsConfiguration"
+    upgradeKymaToCommit
+  fi
+
+  if [ -n "$(kubectl get service -n kyma-system apiserver-proxy-ssl --ignore-not-found)" ]; then
+    log::info "Create DNS Record for Apiserver proxy IP"
+    APISERVER_IP_ADDRESS=$(kubectl get service -n kyma-system apiserver-proxy-ssl -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    APISERVER_DNS_FULL_NAME="apiserver.${DNS_SUBDOMAIN}.${DNS_DOMAIN}"
+    CLEANUP_APISERVER_DNS_RECORD="true"
+    IP_ADDRESS=${APISERVER_IP_ADDRESS} DNS_FULL_NAME=${APISERVER_DNS_FULL_NAME} "${TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS}/create-dns-record.sh"
   fi
 }
 
+# testKyma executes the kyma-testing.sh. labelqueries can be passed as arguments and will be passed to the kyma cli
 function testKyma() {
-    shout "Test Kyma"
-    date
-    "${TEST_INFRA_SOURCES_DIR}"/prow/scripts/kyma-testing.sh
+  testing::inject_addons_if_necessary
+
+  local labelquery=${1}
+  local suitename=${2}
+  local test_args=()
+
+  if [[ -n ${labelquery} ]]; then
+    test_args+=("-l")
+    test_args+=("${labelquery}")
+  fi
+
+  if [[ -n ${suitename} ]]; then
+    test_args+=("-n")
+    test_args+=("${suitename}")
+  fi
+
+  log::banner "Test Kyma " "${test_args[@]}"
+  "${TEST_INFRA_SOURCES_DIR}"/prow/scripts/kyma-testing.sh "${test_args[@]}"
```

Comment:
Can we use kyma cli to run test?

Reply:
kyma-testing.sh uses kyma cli internally. It also adds some convenience functions to the test execution.